File: src/main/java/ru/arc/xserver/repos/RedisRepoMessager.java
package ru.arc.xserver.repos;

import lombok.RequiredArgsConstructor;
import lombok.extern.log4j.Log4j2;
import ru.arc.CommonCore;
import ru.arc.xserver.ChannelListener;
import ru.arc.xserver.RedisManager;

@RequiredArgsConstructor
@Log4j2
public class RedisRepoMessager implements ChannelListener {
    private final RedisRepo<?> redisRepo;
    private final RedisManager redisManager;
    @Override
    public void consume(String channel, String message, String originServer) {
        if(originServer.equals(CommonCore.config.string("server-name","proxy"))) return;
        redisRepo.receiveUpdate(message);
    }

    public void send(String channel, String message){
        redisManager.publish(channel, message);
    }
}

File: src/main/kotlin/ru/arc/xserver/repos/RedisRepoMessager.kt
package ru.arc.xserver.repos

import ru.arc.CommonCore
import ru.arc.xserver.ChannelListener
import ru.arc.xserver.RedisManager

class RedisRepoMessager(
    private val redisRepo: RedisRepo<*>,
    private val redisManager: RedisManager,
) : ChannelListener {
    override fun consume(channel: String, message: String, originServer: String) {
        if (originServer == CommonCore.config?.string("server-name", "proxy")) {
            return
        }
        redisRepo.receiveUpdate(message)
    }

    fun send(channel: String, message: String) {
        redisManager.publish(channel, message)
    }
}
